### ISSUE TITLE
Function wrappers break is-defined checks

### ISSUE THREAD

**Issue by haggholm** [DevExpress/testcafe-hammerhead#1496]
Our site runs some code in a webworker, using an internal package that in some circumstances accesses `localStorage`:

```javascript
if (typeof (localStorage) !== 'undefined') {
  // ...
}
```

However, this completely breaks after Hammerhead (for some reason?) surrounds this with a wrapper.

```javascript
// near the script top:
__get$Storage=function(s){return s}

// later on:
if (typeof __get$Storage(localStorage) !== 'undefined') {
  // ...
}
```

Now, it attempts to evaluate `localStorage` before checking if it exists, so the webworker crashes.

**Comment by miherlosev** [comment]
Hi @haggholm.
Thanks for investigation of this problem.
I've added this problem to current development sprint.

**Comment by MariaRyadchikova** [comment]
Hi @miherlosev ,
Can you give a hint when the issue is planned to be resolved? This thing blocks us to be honest.
Thanks!

**Comment by miherlosev** [comment]
Hi @MariaRyadchikova.
Unfortunately, I can not provide you a strong time estimation when the issue was fixed.
But, I guess it will not be fast. 
Browser raises an exception when function is called with undefined variable in parameter. 
see [sample video](https://www.screencast.com/t/TM5vlThikj).

To fix the issue we need to come up a new way to proxy a `localStorage`.

**Comment by LavrovArtem** [comment]
The fix is published in `testcafe@0.19.1-dev20180305`
